docs/scripts.rst

```diff
@@ -0,0 +1,194 @@
+  Author: Carmen Bianca Bakker <carmenbianca@fsfe.org>
+  Date:   Tue Oct 10 18:27:11 2017 +0200
+
+.. SPDX-SnippetEnd
+
+Year of first commit
+====================
+
+With a custom format just the year of the first commit can be displayed.
+This output is convenient for use in larger scripts.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --reverse --date="format:%Y" --format="format:%cd" | head -n 1
+  2017
+
+.. SPDX-SnippetEnd
+
+*******
+Authors
+*******
+
+Unless the authors have signed away their copyright to a company or the project, the authors are also the copyright holders of their contributions.
+So in a lot of cases it is valuable to know the original authors in order to explictly state these copyright holders.
+
+Commit authors
+==============
+
+Some examples on how to get information about the authors from the codebase.
+
+Based on commit order
+---------------------
+
+Print out the authors known to git in chronological order.
+Awk is used to filter out subsequent duplicate entries so each entry is shown only for the first appearance.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --reverse --all --format="%aN <%aE>" | awk '!seen[$0]++'
+  Carmen Bianca Bakker <carmenbianca@fsfe.org>
+  carmenbianca <carmenbianca@fsfe.org>
+  Carmen Bianca Bakker <carmen@carmenbianca.eu>
+  ...
+
+.. SPDX-SnippetEnd
+
+The same command, but now without the email addresses.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --reverse --all --format="%aN" | awk '!seen[$0]++'
+  Carmen Bianca Bakker
+  carmenbianca
+  Sebastian Schuberth
+  ...
+
+.. SPDX-SnippetEnd
+
+Sorted by name
+--------------
+
+All authors as known to the version control system, simply sorted by name.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --all --format="%aN <%aE>" | sort | uniq
+  Adam Spiers <reuse-tool@adamspiers.org>
+  Ajinkya Patil <ajinkyarangnathpatil@gmail.com>
+  Alvar <8402811+oxzi@users.noreply.github.com>
+  ...
+
+.. SPDX-SnippetEnd
+
+The same command, but now without the email addresses.
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --all --format="%aN" | sort | uniq
+  Adam Spiers
+  Ajinkya Patil
+  Alvar
+  ...
+
+.. SPDX-SnippetEnd
+
+Authors in commit trailers like sign-off
+========================================
+
+A sign-off annotation in a commit also contains author details that can be as valuable.
+
+.. TODO: improve this oneliner based on built-in Git options as documented in https://stackoverflow.com/a/41361273/12013233
+
+.. SPDX-SnippetBegin
+.. SPDX-Snippet-License-Identifier: CC0-1.0
+
+.. code-block:: console
+
+  $ git log --all | grep 'Signed-off-by\|Co-authored-by' | sort | uniq
```

Comment:
May it make sense to make a `grep -i`? I could imagine that there are variations.

Reply:
I see that use-case, even though it is typically handled by automation and therefore properly capitalized. It wouldn't hurt to ignore case here.